src/solver/simulation/hydro-remix-new.cpp

```diff
@@ -0,0 +1,295 @@
+#include "include/antares/solver/simulation/hydro-remix-new.h"
+
+#include <algorithm>
+#include <ranges>
+#include <stdexcept>
+#include <vector>
+
+namespace Antares::Solver::Simulation
+{
+
+int find_min_index(const std::vector<double>& G_plus_H,
+                   const std::vector<double>& new_D,
+                   const std::vector<double>& new_H,
+                   const std::vector<int>& tried_creux,
+                   const std::vector<double>& P_max,
+                   const std::vector<bool>& filter_hours_remix,
+                   double top)
+{
+    double min_val = top;
+    int min_idx = -1;
+    for (int i = 0; i < G_plus_H.size(); ++i)
+    {
+        if (new_D[i] > 0 && new_H[i] < P_max[i] && tried_creux[i] == 0 && filter_hours_remix[i])
+        {
+            if (G_plus_H[i] < min_val)
+            {
+                min_val = G_plus_H[i];
+                min_idx = i;
+            }
+        }
+    }
+    return min_idx;
+}
+
+int find_max_index(const std::vector<double>& G_plus_H,
+                   const std::vector<double>& new_H,
+                   const std::vector<int>& tried_pic,
+                   const std::vector<double>& P_min,
+                   const std::vector<bool>& filter_hours_remix,
+                   double ref_value,
+                   double eps)
+{
+    double max_val = 0;
+    int max_idx = -1;
+    for (int i = 0; i < G_plus_H.size(); ++i)
+    {
+        if (new_H[i] > P_min[i] && G_plus_H[i] >= ref_value + eps && tried_pic[i] == 0
+            && filter_hours_remix[i])
+        {
+            if (G_plus_H[i] > max_val)
+            {
+                max_val = G_plus_H[i];
+                max_idx = i;
+            }
+        }
+    }
+    return max_idx;
+}
+
+static bool operator<=(const std::vector<double>& a, const std::vector<double>& b)
+{
+    std::vector<double> a_minus_b;
+    std::ranges::transform(a, b, std::back_inserter(a_minus_b), std::minus<double>());
+    return std::ranges::all_of(a_minus_b, [](const double& e) { return e <= 0.; });
+}
+
+static bool operator<=(const std::vector<double>& v, const double c)
+{
+    return std::ranges::all_of(v, [&c](const double& e) { return e <= c; });
+}
+
+static bool operator>=(const std::vector<double>& v, const double c)
+{
+    return std::ranges::all_of(v, [&c](const double& e) { return e >= c; });
+}
+
+static void checkInputCorrectness(const std::vector<double>& G,
```

Comment:
Several functions have a lots of args, maybe creating a struct to encapsulate the data will be simpler ?